testdata/channel.go

```diff
@@ -244,7 +250,7 @@ func receive(ch <-chan int) {
 func sender(ch chan int) {
 	for i := 1; i <= 8; i++ {
 		if i == 4 {
-			time.Sleep(time.Microsecond)
+			time.Sleep(time.Millisecond)
```

Comment:
This seems to be long enough to make the test deterministic with the Go toolchain.
But of course it's still racy, if it becomes a problem we can try to find another way using a WaitGroup or something.